### DISCUSSION TITLE
2021-08-03 planning discussion

### DISCUSSION THREAD

**Discussion by kgodey** [mathesar-foundation/mathesar#519] [Maintainer discussions]
### Discussion topics

- **[Retrospective](https://github.com/centerofci/mathesar/discussions/519#discussioncomment-1125135)**: Summary of last week's work.
- **[Active Work](https://github.com/centerofci/mathesar/discussions/519#discussioncomment-1125138)**: Work planned for the next week.
- **[Updates](https://github.com/centerofci/mathesar/discussions/519#discussioncomment-1125203)**: General updates everyone should be aware of.
- **[Questions](https://github.com/centerofci/mathesar/discussions/519#discussioncomment-1125244)**: Questions for the team to answer.

### About
Each Tuesday, we start a discussion about Mathesar's progress. Everyone is welcome to add a new topic to discuss or to comment on existing topics.

**Comment by kgodey** [comment]
## Retrospective

This post lists issue updates from 2021-07-27 until now by milestone.

### [2021-07 improvements](https://github.com/centerofci/mathesar/milestone/28)

- **Issues closed:**
  - [@kgodey]  #485

### [03. Working with Schemas](https://github.com/centerofci/mathesar/milestone/6)

- **Issues closed:**
  - [@andyharley-cci]  #437
  - [@eito-fis]  #365
- **Issues in progress:**
  - [@dhruvkb]  #171
  - [@pavish]  #362
- **New issues created:**
  -  #518, created by @pavish

### [04. Tables from File Import](https://github.com/centerofci/mathesar/milestone/1)

- **Issues closed:**
  - [@obulat]  #491
- **Issues in progress:**
  - [@mathemancer]  #201
- **New issues created:**
  -  #491, created by @kgodey

### [05. Editable Tables](https://github.com/centerofci/mathesar/milestone/4)

- **Issues closed:**
  - [@pavish]  #196
- **Issues in progress:**
  - [@pavish]  #195

### [06. Working with Tables](https://github.com/centerofci/mathesar/milestone/8)

- **Issues closed:**
  - [@andyharley-cci]  #364
  - [@kgodey]  #302
  - [@eito-fis]  #190
- **Issues in progress:**
  - [@eito-fis]  #203
  - [@eito-fis]  #449
  - [@eito-fis]  #397
  - [@ghislaineguerin]  #324
  - [@kgodey]  #326
- **New issues created:**
  -  #515, created by @kgodey
  -  #501, created by @kgodey
  -  #500, created by @kgodey
  -  #499, created by @kgodey
  -  #498, created by @kgodey
  -  #497, created by @kgodey
  -  #496, created by @kgodey
  -  #495, created by @kgodey
  -  #494, created by @kgodey

### [07. Initial Data Types](https://github.com/centerofci/mathesar/milestone/11)

- **Issues in progress:**
  - [@ghislaineguerin]  #239
  - [@powellc]  #387
  - [@mathemancer]  #377
  - [@ghislaineguerin]  #238
  - [@ghislaineguerin]  #234

### [2021-08 improvements](https://github.com/centerofci/mathesar/milestone/29)

- **Issues closed:**
  - [@aaditkamat]  #511
- **Issues in progress:**
  - [@mr-gabe49]  #483
  - [@pavish]  #358
  - [@andyharley-cci]  #225
  - [@Dhruvi16]  #322
- **New issues created:**
  -  #517, created by @eito-fis
  -  #511, created by @aaditkamat
  -  #510, created by @kgodey

### [10. Multiple Database Support](https://github.com/centerofci/mathesar/milestone/5)

- **Issues in progress:**
  - [@aaditkamat]  #395

### [24. Deployment](https://github.com/centerofci/mathesar/milestone/26)

- **New issues created:**
  -  #509, created by @kgodey
  -  #505, created by @kgodey

### [26. Post-MVP](https://github.com/centerofci/mathesar/milestone/27)

- **New issues created:**
  -  #504, created by @kgodey

**Comment by kgodey** [comment]
## Active Work

This post lists active work planned for this week, based on the [Active](https://github.com/centerofci/mathesar/projects/1) GitHub project. It also includes a current summary of milestones involved in active work.

### [2021-08-09] [03. Working with Schemas](https://github.com/centerofci/mathesar/milestone/6)

- **[10 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%2203.%20Working%20with%20Schemas%22) already closed.** 
- **Issues currently in progress:**
  - [@dhruvkb]  #171
  - [@pavish]  #362
- **Issues not yet started:**
  - [@pavish]  #316
  - [@dhruvkb]  #518

### [2021-08-11] [04. Tables from File Import](https://github.com/centerofci/mathesar/milestone/1)

- **[26 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%2204.%20Tables%20from%20File%20Import%22) already closed.** 
- **Issues currently in progress:**
  - [@mathemancer]  #201
- **Issues not yet started:**
  - [@andyharley-cci]  #421
  - [@pavish]  #296
  - [@pavish]  #295

### [2021-08-16] [05. Editable Tables](https://github.com/centerofci/mathesar/milestone/4)

- **[6 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%2205.%20Editable%20Tables%22) already closed.** 
- **Issues currently in progress:**
  - [@pavish]  #195
- **Issues not yet started:**
  - [@pavish]  #197

### [2021-08-23] [06. Working with Tables](https://github.com/centerofci/mathesar/milestone/8)

- **[9 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%2206.%20Working%20with%20Tables%22) already closed.** 
- **Issues currently in progress:**
  - [@eito-fis]  #203
  - [@eito-fis]  #449
  - [@eito-fis]  #397
  - [@ghislaineguerin]  #324
  - [@kgodey]  #326
- **Issues not yet started:**
  - [@eito-fis]  #452
  - [@eito-fis]  #494
  - [@eito-fis]  #501
  - [@eito-fis]  #515
- **Issues remaining in backlog:**
  - 14 [frontend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20frontend%22+milestone%3A%2206.%20Working%20with%20Tables%22) issues

### [2021-08-30] [07. Initial Data Types](https://github.com/centerofci/mathesar/milestone/11)

- **[2 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%2207.%20Initial%20Data%20Types%22) already closed.** 
- **Issues currently in progress:**
  - [@ghislaineguerin]  #239
  - [@powellc]  #387
  - [@mathemancer]  #377
  - [@ghislaineguerin]  #238
  - [@ghislaineguerin]  #234
- **Issues not yet started:**
  - [@ghislaineguerin]  #240
  - [@powellc]  #388
  - [@powellc]  #247
  - [@ghislaineguerin]  #235
  - [@ghislaineguerin]  #236
  - [@ghislaineguerin]  #237
  - [@ghislaineguerin]  #241
  - [@mathemancer]  #378
  - [@mathemancer]  #379
  - [@mathemancer]  #380
  - [@mathemancer]  #381
  - [@mathemancer]  #382
  - [@mathemancer]  #383
  - [@kgodey]  #367
  - [@kgodey]  #369
  - [@kgodey]  #368
  - [@rootxnet]  #244
  - [@mathemancer]  #434
- **Issues remaining in backlog:**
  - 37 [backend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20backend%22+milestone%3A%2207.%20Initial%20Data%20Types%22) and [database](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20database%22+milestone%3A%2207.%20Initial%20Data%20Types%22) issues
  - 8 [frontend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20frontend%22+milestone%3A%2207.%20Initial%20Data%20Types%22) issues

### [2021-08-31] [2021-08 improvements](https://github.com/centerofci/mathesar/milestone/29)

- **[4 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%222021-08%20improvements%22) already closed.** 
- **Issues currently in progress:**
  - [@mr-gabe49]  #483
  - [@pavish]  #358
  - [@andyharley-cci]  #225
  - [@Dhruvi16]  #322
- **Issues not yet started:**
  - [@mathemancer]  #280
- **Issues remaining in backlog:**
  - 2 [backend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20backend%22+milestone%3A%222021-08%20improvements%22) and [database](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20database%22+milestone%3A%222021-08%20improvements%22) issues
  - 9 [frontend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20frontend%22+milestone%3A%222021-08%20improvements%22) issues
  - 6 [design](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20design%22+milestone%3A%222021-08%20improvements%22) issues
  - 1 [infrastructure](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20infrastructure%22+milestone%3A%222021-08%20improvements%22) issues

### [2021-09-06] [10. Multiple Database Support](https://github.com/centerofci/mathesar/milestone/5)

- **[6 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%2210.%20Multiple%20Database%20Support%22) already closed.** 
- **Issues currently in progress:**
  - [@aaditkamat]  #395
- **Issues remaining in backlog:**
  - 4 [backend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20backend%22+milestone%3A%2210.%20Multiple%20Database%20Support%22) and [database](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20database%22+milestone%3A%2210.%20Multiple%20Database%20Support%22) issues
  - 4 [frontend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20frontend%22+milestone%3A%2210.%20Multiple%20Database%20Support%22) issues
  - 2 [design](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20design%22+milestone%3A%2210.%20Multiple%20Database%20Support%22) issues

**Comment by kgodey** [comment]
## Updates

### Roadmap Updates
The roadmap has been updated to account for editing records taking more time than originally anticipated.
- Please review the [new order and deadlines](https://github.com/centerofci/mathesar/milestones?direction=asc&sort=due_date&state=open).
- Feel free to participate in the discussion about how to approach editing: #514

### Presentation
The Mathesar team is presenting on progress so far to the larger CCI research meeting. Please let me know if you'd like an invite. It's at 11AM Eastern Time on Thursday, Aug 5.

### Contributors
- We have two new contributors this week with merged PRs. Thank you, @aaditkamat and @obulat!
- I'd also like to mention @dhruvkb's work on #513, which involved implementing a whole new page in Mathesar and is almost merged.
- @andyharley-cci got his first backend PR merged today. :)

### Contractors
- @powellc is working with us part-time on backend work from now until the end of October. Welcome, @powellc!
- @rootxnet is working with us on #244 and its sub-issues. Welcome, @rootxnet!

Comments or feedback on any of the above is welcome.

**Comment by kgodey** [comment]
## Questions

@mathemancer, @eito-fis, @ghislaineguerin, @pavish: Your responses are required. Others are also welcome to respond.

### Retrospective
- What went well this week?
- What could have gone better this week?
- Are there any improvements to our process you'd like to suggest?

### Active Work
- Does your workload for this week look okay?
- Are there any blockers to meeting our upcoming deadlines?

### Discussion
- Any feedback on the weekly discussion (either format or content)?
- Anything else you'd like to add?

**Reply by kgodey** [reply]
> What went well this week?

- I got to actually write some code for #488  
- It's nice that we have more contributors.

> What could have gone better this week?

Backend work is a little slower than I'd like. I'd like to try to find more time to help with it.

> Are there any improvements to our process you'd like to suggest?

No.

> Does your workload for this week look okay?

Yes. I'm also going to be speccing out more issues, which isn't represented in the active work.

> Are there any blockers to meeting our upcoming deadlines?

Not that I'm aware of.

> Any feedback on the weekly discussion (either format or content)?

I'm hoping that putting the questions in one post will make it easier for people to find them. I wasn't getting a lot of responses when they were at the end of other long posts.

**Reply by pavish** [reply]
> What went well this week?

* I'm glad we got new contributors, and for the extra frontend help (Thanks @dhruvkb !).
* This week was not very productive for me. I spent a lot of time thinking on ways to handle editing with the current data loading approach, before posting it for suggestions.

> What could have gone better this week?

* It would have been better if we had planned the data loading strategy earlier, considering all our features.
* I still find that it's okay, we do not have a lot of dependent features on top of spreadsheet, so changes should be simple.
* I would like to hit some milestones and speed up frontend.

> Are there any improvements to our process you'd like to suggest?

None, at the moment.

> Does your workload for this week look okay?

Yes. It looks good to me.

> Are there any blockers to meeting our upcoming deadlines?

I do not see any blockers.

> Any feedback on the weekly discussion (either format or content)?
> Anything else you'd like to add?

None.

**Reply by mathemancer** [reply]
> What went well this week?

Data Checking API flow is finally done!

> What could have gone better this week?

I want to get faster with PR reviews; will be focusing on that.

> Are there any improvements to our process you'd like to suggest?

None

> Does your workload for this week look okay?

I think it looks good.

> Are there any blockers to meeting our upcoming deadlines?

Not to my knowledge.

> Any feedback on the weekly discussion (either format or content)?

I think the new format is working well.

> Anything else you'd like to add?

Nope!

**Reply by powellc** [reply]
> * What went well this week?

Starting to get more comfortable with the codebase and got a few PR reviews under my belt and some data type tests run successfully :D

> * What could have gone better this week?

Need to find a good rhythm for getting ticket work done and communicated out. Still a little new to everything

> * Are there any improvements to our process you'd like to suggest?

Nope!

> * Does your workload for this week look okay?

Yup, going to finish up reviewing the current boolean type implementation and move on to adding filtering on booleans.

> * Are there any blockers to meeting our upcoming deadlines?

Nope.

> * Any feedback on the weekly discussion (either format or content)?

Love it!

> * Anything else you'd like to add?

Nada 😄

**Reply by eito-fis** [reply]
> What went well this week?

Finally getting some PR reviews in!

> What could have gone better this week?

Progress on some easier PRs was slow - would like to get the PR right in fewer review cycles if it's simple.

> Are there any improvements to our process you'd like to suggest?

N / A

>  Does your workload for this week look okay?

Seems fine to me.

> Are there any blockers to meeting our upcoming deadlines?

N / A

> Any feedback on the weekly discussion (either format or content)?

Having all questions in once place is definitely nice.

> Anything else you'd like to add?

N / A

**Comment by ghislaineguerin** [comment]
> What went well this week?

I had been on vacation, so not an entire week. But it was nice to be back and catch up quickly as we do a lot of async communication. 
I also liked how we planned and practiced for the presentation. It felt like a very productive meeting.

> What could have gone better this week?

It would have been easier if I didn't have so many reviews open. Closing them has been challenging as many sub-parts for data types are still not defined or become more evident as I add new parts.

> Are there any improvements to our process you'd like to suggest?

I should try to identify those parts before I start designing and plan to include them in my planning.

> Does your workload for this week look okay?

It does.

> Are there any blockers to meeting our upcoming deadlines?

No blockers.

> Any feedback on the weekly discussion (either format or content)?

No, I think it works well.

> Anything else you'd like to add?

Not for now.
